### ISSUE TITLE
Karten anzeigen

### ISSUE THREAD

**Issue by lindschoki** [JirkaDellOro/EIA2#214]
Hallo, ich hab ein Problem bei der Karten-Erzeugung:

in meinem interface habe ich meinen Kartenobjekten einen Farbwert und einen Inhaltswert gegeben. Wie muss ich nun meine Funktion um die Karten zu generieren aufbauen, damit nach math.random die Karten auch mit unterschiedlichen Farben und Inhalten erzeugt werden, weis hier leider nicht weiter.

Vielen Dank im Voraus

**Comment by JirkaDellOro** [comment]
Hallo Arno,

das Konzept ist noch das alte und der Code ist noch nicht im Remote-Repository. Ich helfe gerne, wenn ich mehr Informationen habe.

**Comment by JirkaDellOro** [comment]
Aber folgende Info kann ich schon geben: es ist nicht zwingend erforderlich, die das Kartendeck, aus dem die Handkarten schließlich ausgewählt werden, prozedural zu generieren. Sie können auch literal angegeben werden.

**Comment by lindschoki** [comment]
Also würde es reichen wenn in einer div box zum beispiel "7" und "rot" steht?

**Comment by JirkaDellOro** [comment]
Die Farbe sollte schon als Hintergrundfarbe erscheinen und nicht als Text angegeben werden müssen. Liegt hier das Problem?

**Comment by lindschoki** [comment]
ja ich weis nicht wie das funktioniert, ich hab auch noch mal ein neues Konzept in meinem Repository hochgeladen.

**Comment by JirkaDellOro** [comment]
Ok, habe ich gesehen. Sieht schon etwas besser aus. x sollte anders heißen, damit man weiß, was es bedeuten soll.

Man kann das Problem so lösen, wie ich es bei den Boxes aus didaktischen Gründen gemacht habe, aber ich rate davon ab. Besser ist es nämlich, hierfür CSS zu nutzen und bei den neu erstellten divs einfach entsprechende Selektor-Attribute zu definieren, auf die sich das Stylesheet dann beziehen kann.

Ist das verständlich?

**Comment by JirkaDellOro** [comment]
... deswegen kann ich keine Aussage zur Positionierung treffen.

**Comment by lindschoki** [comment]
Ok trotzdem vielen Dank.

**Comment by JirkaDellOro** [comment]
Ich hatte eine andere Reaktion erwartet. So etwas wie: "Oh, Verzeihung, meine Skizze liefere ich sofort"

**Comment by JirkaDellOro** [comment]
Ich bin morgen früh wieder da... gute Nacht.

**Comment by lindschoki** [comment]
Hatte gedacht das es vielleicht schon etwas spät ist und wollte Sie daher auch nicht weiter stören. Kam vielleicht falsch rüber.

**Comment by JirkaDellOro** [comment]
Ja, kam anders rüber. Und ein Vorteil der asynchronen Kommunikation ist, dass man unterbrechen kann und teilnehmen, wenn es zeitlich passt. Daher stellen Postings keine Störung dar. Außerdem können andere  mit reinkommen und ggf. die Hilfestellung weiterführen.

Eine Zeichnung ist jetzt da, wie ich sehe. Und so brutal simpel sie auch ist, klärt sie doch, dass eine ausgefeilte Positionierung nicht erforderlich scheint. Die Handkarten sollen offensichtlich einfach horizontal aufgereiht werden, das ist das Standardverhalten von inline-Elementen in HTML. Zu empfehlen ist aber, die Handkarten in einem separierten Bereich auf der Seite abzulegen. Siehe Aufgabenstellung: "Auch eine grundsätzliche Aufteilung von Spielbereichen für Handkarten, Ablage- und Nachziehstapel können in der HTML-Datei definiert sein."

Bleibt also nur noch die Farbgebung. Die Farben könnte man im Stylesheet definieren und mit unterschiedlichen Selektoren verknüpfen. Bei der Erzeugung der Handkarten gibt man den entsprechenden Selektor oder die Selektoren als Attribut(e) mit. Das ist die sauberste Lösung, da wir schließlich ein System nutzen, das Formatvorlagen anbietet. Alternativ könnte man die Farbe einfach auf der Element-Ebene definieren, da es doch kaskadierende Formatvorlagen sind. Also z.B. div.style.backgroundColor = ....

**Comment by JirkaDellOro** [comment]
Noch ein Tipp zum Code: statt 108 Variablen zu definieren um deren Werte dann in ein Array zu kopieren, kann man diese gleich bei der Definition des Arrays angeben, das wird kompakter.

let cards: UNO_Card[] = [ 
   {color: "#ff0000", value: "0"}, {color: "#ff0000", value: "1"}, ...
];

Ich habe mir hier erlaubt, die Bezeichner etwas auf Vordermann zu bringen. Beachten Sie unbedingt den CodingStyle (siehe MindMap)

**Comment by lindschoki** [comment]
Vielen Dank für die Tipps, ich werde versuchen Sie umzusetzen.
